File: Programmers/연습문제/Level 1/최소직사각형/Solution.js
function solution(sizes) {
    const [max_w, max_h] = getMaxLengths(sizes);
    const answer = getCardSize(max_w, max_h);
    return answer;
}

function getMaxLengths(sizes) {
    let max_w = 0;
    let max_h = 0;
    for (let [w, h] of sizes) {
        [w, h] = getSorted(w, h);
        max_w = max_w < w ? w : max_w;
        max_h = max_h < h ? h : max_h;
    }
    return [max_w, max_h];
}

function getSorted(l1, l2) {
    return [l1, l2].sort((a, b) => (a-b));
}

function getCardSize(w, h) {
    return w*h;
}


// TC
const sizes = [[60, 50], [30, 70], [60, 30], [80, 40]];
console.log(solution(sizes));

File: Programmers/연습문제/Level 1/최소직사각형/Solution.swift
import Foundation

func solution(_ sizes:[[Int]]) -> Int {
    var (max_w, max_h) = (0, 0)
    for size in sizes {
        max_w = max(max_w, size.max()!)
        max_h = max(max_h, size.min()!)
    }
    return max_w * max_h
}

    
// TC
let sizes = [[60, 50], [30, 70], [60, 30], [80, 40]]
print(solution(sizes))
